util/smaractconfig.py

```diff
@@ -0,0 +1,160 @@
+                            channel, smaract.SA_CTLDLL.SA_CTL_REF_OPT_BIT_AUTO_ZERO)  # auto zero
+
+    # Run the referencing
+    logging.info("Starting reference...")
+    dev.reference().result()
+    time.sleep(0.1)
+    logging.info("Complete")
+
+    # Set normal referencing mode
+    for channel in  channels:
+        logging.info("Setting the reference mode to normal for channel %d", channel)
+        dev.SetProperty_i32(smaract.SA_CTLDLL.SA_CTL_PKEY_REFERENCING_OPTIONS,
+                            channel, smaract.SA_CTLDLL.SA_CTL_REF_OPT_BIT_NORMAL)  # normal
+
+
+def set_zero(dev, channels):
+    """
+    Using the scale offset, set the current position of the controller to 0.
+    """
+    if not channels:
+        return
+
+    logging.info("Setting scale offsets. Current position: %s", dev.position.value)
+
+    for channel in channels:
+        logging.info("Setting the scale offset for channel %d", channel)
+        scale_offset = dev.GetProperty_i64(smaract.SA_CTLDLL.SA_CTL_PKEY_LOGICAL_SCALE_OFFSET, channel)
+        position = dev.GetProperty_i64(smaract.SA_CTLDLL.SA_CTL_PKEY_POSITION, channel)
+        # set the new scale offset
+        dev.SetProperty_i64(smaract.SA_CTLDLL.SA_CTL_PKEY_LOGICAL_SCALE_OFFSET, channel, scale_offset + position)
+
+    dev._updatePosition()
+    logging.info("New position: %s", dev.position.value)
+    logging.info("Complete.")
+
+
+def main(args):
+
+    parser = argparse.ArgumentParser(prog="smaractconfig",
+                                     description='Run a one time config for the SmarAct ')
+
+    parser.add_argument('--locator', required=True, type=str, help='Run the operations with the simulator', action="store")
+
+    parser.add_argument("--log-level", dest="loglev", metavar="<level>", type=int,
+                        default=1, help="set verbosity level (0-2, default = 1)")
+
+    parser.add_argument('--calibrate', metavar='N', type=int, nargs='+',
+                    help="Run the calibration command. Specify the channel to calibrate (0, 1, 2)")
+
+    parser.add_argument('--autozero', metavar='N', type=int, nargs='+',
+                    help="Run referencing with autozero functionality. Specify the channel to calibrate (0, 1, 2)")
+
+    parser.add_argument('--setzero', metavar='N', type=int, nargs='+',
+                    help="Force the current position on a channel to be 0 by setting the scale offset. Specify the channel to calibrate (0, 1, 2)")
+
+    options = parser.parse_args(args[1:])
+
+    # Set up logging before everything else
+    if options.loglev < 0:
+        logging.error("Log-level must be positive.")
+        return 127
+    loglev_names = (logging.WARNING, logging.INFO, logging.DEBUG)
+    loglev = loglev_names[min(len(loglev_names) - 1, options.loglev)]
+    logging.getLogger().setLevel(loglev)
+
+    if options.locator:
+        CONFIG_3DOF['locator'] = options.locator
+
+    dev = smaract.MCS2(**CONFIG_3DOF)
```

Comment:
You can write `smaract.MCS2(locator=locator, axes=axes, **CONFIG_3DOF)`
And you should compute axes just here, based on which axes are specified in the options.